op-node/rollup/derive/channel.go

```diff
@@ -147,8 +151,41 @@ func (ch *Channel) Reader() io.Reader {
 // Warning: the batch reader can read every batch-type.
 // The caller of the batch-reader should filter the results.
 func BatchReader(r io.Reader) (func() (*BatchData, error), error) {
+	// Buffer to store the data for reuse
+	var buffer strings.Builder
+	if _, err := io.Copy(&buffer, r); err != nil {
+		return nil, err
+	}
+
+	tmpReader := strings.NewReader(buffer.String())
+	// Read the first byte to be used to determine the compression type
+	compressionType := make([]byte, 1)
+	if _, err := tmpReader.Read(compressionType); err != nil {
+		return nil, err
+	}
+
+	// Reset the reader with the original data
+	r = strings.NewReader(buffer.String())
+	var reader func(io.Reader) (io.Reader, error)
+	// If the last 4 bits is 8, then it is a zlib reader
+	if compressionType[0]&0x0F == 8 {
+		reader = func(r io.Reader) (io.Reader, error) {
+			return zlib.NewReader(r)
+		}
+		// If the bits equal to 1, then it is a brotli reader
+	} else if compressionType[0]&0x0F == 1 {
+		// remove the first byte by reading it
+		_, err := r.Read(make([]byte, 1))
+		if err != nil {
+			return nil, err
+		}
+		reader = func(r io.Reader) (io.Reader, error) {
+			return brotli.NewReader(r), nil
+		}
+	}
```

Comment:
Is an invalid version byte handled properly?  (Should result in the batch being ignored.).   Please make sure there's a test for this case.